File: String_Functions/Palindrome_Check/palindrome_check.py
class PalindromeCheck:
    def reverse_string(self, string):
        length = len(string)  # length of the string
        reversed_string = ""
        for i in range(length - 1, -1, -1):
            reversed_string += string[i]  # we reverse the string by adding characters from the end of the string.
        return reversed_string

    def check_for_palindrome(self, string):  # checking for palindrome
        reversed_string = self.reverse_string(string)  # reversing the string by calling previously defined method "reverse_string"
        return string.lower() == reversed_string.lower()  # compare ignoring case

if __name__ == "__main__":
    obj = PalindromeCheck()
    example_palindrome = "racecar"
    print(obj.check_for_palindrome(example_palindrome))
    print(obj.check_for_palindrome("NotAPalindrome"))

File: String_Functions/Palindrome_Check/palindrome_check.cpp
#include <iostream>
#include <string>

class PalindromeCheck {
public:
    std::string reverseString(const std::string& str) {
        std::string reversed;
        for (int i = str.length() - 1; i >= 0; i--) {
            reversed += str[i]; // we reverse the string by adding characters from the end of the string.
        }
        return reversed;
    }

    bool checkForPalindrome(const std::string& str) { // checking for palindrome
        std::string reversed = reverseString(str); // reversing the string by calling previously defined method "reverseString"
        return str.compare(reversed) == 0; // compare returns 0 if both strings are identical
    }
};

int main() {
    PalindromeCheck obj;
    std::string examplePalindrome = "racecar";
    std::cout << std::boolalpha << obj.checkForPalindrome(examplePalindrome) << std::endl;
    std::cout << std::boolalpha << obj.checkForPalindrome("NotAPalindrome") << std::endl;

    return 0;
}

